### PR TITLE
Feat/core 115

### PR DESCRIPTION
**Pull request by smancebo**

### CODE REVIEW

File: env.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+export default {
```
Comment (SparoHawk) [inline env.ts:1]: Are we going to save this directly in the repo or should we use real `.env`?

File: src/modules/users/index.ts [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+/**
+ * @description Kanvas Users Module
+ */
+export default class Users extends Base {
+  private baseUrl: string
+
+  constructor(http: HttpClient) {
+    super(http);
+    this.baseUrl = '/users';
```
Comment (SparoHawk) [inline src/modules/users/index.ts:25]: **praise:** This is a nice idea! 👍🏻 I will extend this to the Auth and Application Settings modules. 😎

File: src/modules/users/index.ts [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+    const params = { page, limit, sort, format };
+
+    if (format) {
+      const { data } = await this.http.request<KanvasFormatedResponse<UserInterface>>({
+        method: 'GET',
+        url: this.baseUrl,
+        params,
+      });
+      return data;
+    }
```
Comment (SparoHawk) [inline src/modules/users/index.ts:87]: **thought:** Should we try to make this _Pagination_ implementation some kind of default behavior of the HttpClient? 🤔

File: src/modules/users/index.ts [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+
+    return data;
+  }
+
+  /**
+   * @description Delete an user by its ID
+   * @param {number} userId 
+   * @returns {Promise<UserInterface>} - Deleted user
+   */
+  async delete(userId: number): Promise<UserInterface> {
```
Comment (SparoHawk) [inline src/modules/users/index.ts:115]: **question(non-blocking):** Shouldn't we shy away from using JavaScript key/reserved words as method/variable names? 🚯

File: src/types/user.interface.ts [hunk 5]
```diff
@@ -0,0 +1,63 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
```
Comment (SparoHawk) [inline src/types/user.interface.ts:1]: **thought(non-blocking):** Users has a lot of fields that were inherited from an earlier attempt at a core API. I think we should focus on only interfacing the things Kanvas needs to operate properly. What do you think? 💭

**Review by SparoHawk — COMMENTED** [review]
**praise:** Overall very nice ideas an implementations. I left a few comments that you should check out. Otherwise awesome job! 👍🏻

**Review by SparoHawk — APPROVED** [review]

**Comment by SparoHawk** [comment]
Oppps! I hit comment instead of approve. 🦡

File: env.ts (hunk 1 above)
Comment (smancebo) [reply env.ts:1]: Made this refactor to has an unique entry point for envs

File: src/modules/users/index.ts (hunk 3 above)
Comment (smancebo) [reply src/modules/users/index.ts:87]: lets discuss this

File: src/modules/users/index.ts (hunk 4 above)
Comment (smancebo) [reply src/modules/users/index.ts:115]: Agree

File: src/types/user.interface.ts (hunk 5 above)
Comment (smancebo) [reply src/types/user.interface.ts:1]: Agree, lets discuss the fields that we are going to keep on User interface

File: env.ts (hunk 1 above)
Comment (SparoHawk) [reply env.ts:1]: I understand this refactor. My question alludes as the whether we should expose this kind of data to the repo or not.
And if we should use `.env` that adds the variables to the `process.env` of Node or not. 💮

**Pull request merged**